.github/workflows/c-cpp.yml

```diff
@@ -95,15 +103,15 @@ jobs:
       runs-on: ${{ matrix.config.os }}
 
       steps:
-        - uses: actions/checkout@v3
+        - uses: actions/checkout@v4
           with:
             submodules: recursive
 
         - if: startsWith(matrix.config.os, 'ubuntu')
           run: |
             sudo sed -i 's/azure\.//' /etc/apt/sources.list
             sudo apt-get update
-            sudo apt-get install -y clang xorg-dev libxinerama-dev libxcursor-dev libgles2-mesa-dev libegl1-mesa-dev libglfw3-dev libglew-dev libstdc++-12-dev
+            sudo apt-get install -y clang emscripten xorg-dev libxinerama-dev libxcursor-dev libgles2-mesa-dev libegl1-mesa-dev libglfw3-dev libglew-dev libstdc++-12-dev
```

Comment:
Please add a separate `if` block to install `empscripten` only for corresponding build runs.

Reply:
Ok i will change the install setup.

Reply:
Please keep the original `if`-block condition intact. You can change the name of the new build configuration to `Emscripten (Ubuntu)` to avoid prefix collisions - this would be similar to `Android (Ubuntu)`.

Reply:
I applied the changes, i'm thinking now that probably for the Emscripten compilation is not required some packages, because Emscripten is using its internal libraries.

Reply:
@corporateshark instead to create an if condition, it should be possible create two `jobs` :

```
install_ubuntu_packages:  |
            sudo sed -i 's/azure\.//' /etc/apt/sources.list
            sudo apt-get update
            sudo apt-get install -y clang xorg-dev libxinerama-dev libxcursor-dev libgles2-mesa-dev libegl1-mesa-dev libglfw3-dev libglew-dev libstdc++-12-dev
install_emscripten_package: |
            sudo apt-get install -y emscripten
```
inside Build for Ubuntu:
`uses: install_ubuntu_packages`
inside Build for Emscripten:
`uses: install_emscripten_package`

just an idea to test.

Reply:
> @corporateshark instead to create an if condition, it should be possible create two `jobs` :
> 
> ```
> install_ubuntu_packages:  |
>             sudo sed -i 's/azure\.//' /etc/apt/sources.list
>             sudo apt-get update
>             sudo apt-get install -y clang xorg-dev libxinerama-dev libxcursor-dev libgles2-mesa-dev libegl1-mesa-dev libglfw3-dev libglew-dev libstdc++-12-dev
> install_emscripten_package: |
>             sudo apt-get install -y emscripten
> ```
> 
> inside Build for Ubuntu: `uses: install_ubuntu_packages` inside Build for Emscripten: `uses: install_emscripten_package`
> 
> just an idea to test.

This can be a separate PR if you want.